src/Mutation/CommentCreate.php

```diff
@@ -35,15 +35,17 @@ public static function register_mutation() {
 	public static function get_input_fields() {
 		return [
 			'commentOn'   => [
-				'type'        => 'Int',
+				'type'        => [
+					'non_null' => 'ID',
+				],
 				'description' => __( 'The ID of the post object the comment belongs to.', 'wp-graphql' ),
 			],
 			'author'      => [
-				'type'        => 'String',
+				'type'        => boolval( get_option( 'require_name_email' ) ) ? [ 'non_null' => 'String' ] : 'String',
```

Comment:
If an option can make this nullable, we should probably leave it nullable in the Schema. 

Otherwise setting/unsetting this option in the admin would be a breaking change for clients.

Reply:
That doesn't make sense to me. 

When this option is enabled on the server any mutations not passing this data will start breaking either way, right?. The only difference is whether it's a schema error or a server error (not sure if those are the correct terms 🤷 ). If this remains, then when this option is enabled the schema will correctly indicate to devs what the server requires, which seems extremely helpful. 

If this option is disabled, It can still accept name/email but it isn't required...i don't see what would break...am I missing something?

Reply:
@moonmeister the difference as I see it is that a schema change can _break_ the frontend (possibly even preventing it from compiling) whereas if it fails validation on the server, an existing frontend will just handle it like any other error response.

Reply:
Personally if someone flips that switch and causes something to break I'd rather my build break than finding out in production.

Reply:
What exactly do you fear would break? The UserError you added will ensure that sites which require the param submit it, while keeping the schema non-breaking.

Reply:
Okay, We don't want to change this. Keep in `nullable`